test-data/unit/check-classes.test

```diff
@@ -6031,7 +6031,7 @@ class A:
     def __new__(cls) -> int:  # E: Incompatible return type for "__new__" (returns "int", but must return a subtype of "A")
         pass
 
-reveal_type(A())  # N: Revealed type is 'builtins.int'
+reveal_type(A())  # N: Revealed type is '__main__.A'
```

Comment:
I think we should keep the declared return type for the class itself. The original PR closed an issue called "Honor return type of `__new__`", so this change would be a step back.

Reply:
We report an error at the `__new__` declaration site, and there are a bunch of other situations where we wouldn't honor it before (like if it is a union or something other than an instance or tuple), so I think this actually makes thing a little more consistent.